### ISSUE TITLE
The `LocalPartitionQueue::producerPromises_` is no longer needed

### ISSUE THREAD

**Issue by lingbin** [facebookincubator/velox#6387]
### Description

`producerPromises_` is used by the producer (`LocalPartition`) to be notified after all data has been consumed by `LocalExchange`.  But after #4385, `LocalPartition` no longer waits for `LocalExchange` to drain all the data. This also means that after all the data in the queue has been consumed, there is no need to notify the "producer" at all.

So, `LocalPartitionQueue::producerPromises_` is no longer needed, we should remove it, its notification to the producer is redundant, right?

I can submit a PR to fix it.
